frb/scripts/r_vs_dm.py

```diff
@@ -0,0 +1,187 @@
+""" Estimate magnitude range for a host FRB given DM """
```

Comment:
does this need to be a separate script from 

https://github.com/FRBs/FRB/blob/main/frb/scripts/pz_dm.py 

??

I suspect not.  If not, please merge this into that script.

Also, please place the underlying functions in here:

https://github.com/FRBs/FRB/tree/main/frb/galaxies

Reply:
@profxj , changes made, and code rearranged as discussed. Also the z-range plot now is color gradient to illustrate the PDF of redshift (instead of a single color in the 95% CI band)